internal/scheduler/nodedb/nodedb.go

```diff
@@ -556,59 +557,52 @@ func BindPodToNode(req *schedulerobjects.PodRequirements, node *schedulerobjects
 	return node, nil
 }
 
-// EvictPodFromNode returns a copy of node with req evicted from it. Specifically:
-// - The job is marked as evicted on the node.
-// - AllocatedByJobId and AllocatedByQueue are not updated.
-// - Resources requested by the evicted pod are marked as allocated at priority evictedPriority.
-func EvictPodFromNode(req *schedulerobjects.PodRequirements, node *schedulerobjects.Node) (*schedulerobjects.Node, error) {
-	jobId, err := JobIdFromPodRequirements(req)
-	if err != nil {
-		return nil, err
-	}
-	queue, err := QueueFromPodRequirements(req)
-	if err != nil {
-		return nil, err
-	}
-	node = node.DeepCopy()
-
-	// Ensure we track allocated resources at evictedPriority.
-	if _, ok := node.AllocatableByPriorityAndResource[evictedPriority]; !ok {
-		pMin := int32(math.MaxInt32)
-		ok := false
-		for p := range node.AllocatableByPriorityAndResource {
-			if p < pMin {
-				pMin = p
-				ok = true
-			}
-		}
-		if ok {
-			node.AllocatableByPriorityAndResource[evictedPriority] = node.AllocatableByPriorityAndResource[pMin].DeepCopy()
-		}
-	}
-
+// EvictJobFromNodeInPlace marks the given job as evicted and moves its resources from the job's
+// priority to preemptedPriority; it does not update AllocatedByJobId and AllocatedByQueue.
+func EvictJobFromNodeInPlace(priorityClasses map[string]configuration.PriorityClass, job interfaces.LegacySchedulerJob, node *schedulerobjects.Node) error {
+	jobId := job.GetId()
 	if _, ok := node.AllocatedByJobId[jobId]; !ok {
-		return nil, errors.Errorf("job %s has no resources allocated on node %s", jobId, node.Id)
+		return errors.Errorf("job %s has no resources allocated on node %s", jobId, node.Id)
 	}
+
+	queue := job.GetQueue()
 	if _, ok := node.AllocatedByQueue[queue]; !ok {
-		return nil, errors.Errorf("queue %s has no resources allocated on node %s", queue, node.Id)
+		return errors.Errorf("queue %s has no resources allocated on node %s", queue, node.Id)
 	}
+
 	if node.EvictedJobRunIds == nil {
 		node.EvictedJobRunIds = make(map[string]bool)
 	}
 	if _, ok := node.EvictedJobRunIds[jobId]; ok {
-		// TODO: We're using run ids instead of job ids for now.
```

Comment:
As far as I can tell we are actually using job IDs here. On `master`, the value of `jobId` comes from `JobIdFromPodRequirements`:

https://github.com/armadaproject/armada/blob/a678bb59a73f7dfe030f5ce62059a006208f09f8/internal/scheduler/nodedb/nodedb.go#L564

… which gets it from the job ID annotation:

https://github.com/armadaproject/armada/blob/a678bb59a73f7dfe030f5ce62059a006208f09f8/internal/scheduler/nodedb/nodedb.go#L674

… and that's set to the job ID:

https://github.com/armadaproject/armada/blob/a678bb59a73f7dfe030f5ce62059a006208f09f8/internal/scheduler/common.go#L263

Reply:
Yes, that's right. In the current system, the executors (more specifically, Kubernetes) generate the run ids. As a result, the scheduler has no clue what they are and we use jobs ids instead. In the Pulsar-backed scheduler, the scheduler creates the run ids and we'll be able to switch to using run ids here.